src/PostgREST/ApiRequest.hs

```diff
@@ -48,7 +49,7 @@ type RequestBody = BL.ByteString
 -- | Types of things a user wants to do to tables/views/procs
 data Action = ActionCreate | ActionRead
             | ActionUpdate | ActionDelete
-            | ActionInfo   | ActionInvoke
+            | ActionInfo   | ActionInvoke{isReadOnly :: Bool}
```

Comment:
Don't know if it's feasible, but I have an idea for a simplification. What if:

* ActionInvoke went back to not taking the isReadOnly option
* iRpcQParams became iRpcParams which ApiRequest.hs would fill in from either the post body or query params as relevant
* Make a new query attribute iReadOnly and set it true for ActionRead, ActionInfo, ActionInspect, or GET-based ActionInvoke. Refer to iReadOnly in transactionMode

Do you think that would be a net simplification, or just shift complexity around? I'm feeling there would be less pattern matching in other places that way.